lib/streamlit/delta_generator.py

```diff
@@ -368,18 +368,29 @@ def __deepcopy__(self, _memo):
         return dg
 
     @property
-    def _parent_block_types(self) -> ParentBlockTypes:
-        """Iterate all the block types used by this DeltaGenerator and all
-        its ancestor DeltaGenerators.
-        """
+    def _anscestors(self) -> Iterable["DeltaGenerator"]:
         current_dg: DeltaGenerator | None = self
         while current_dg is not None:
-            if current_dg._block_type is not None:
-                yield current_dg._block_type
+            yield current_dg
             current_dg = current_dg._parent
 
-    def _count_num_of_parent_columns(self, parent_block_types: ParentBlockTypes) -> int:
-        return sum(1 for parent_block in parent_block_types if parent_block == "column")
+    @property
+    def _anscestor_block_types(self) -> AnscestorBlockTypes:
+        """Iterate all the block types used by this DeltaGenerator and all
+        its ancestor DeltaGenerators.
+        """
+        for a in self._anscestors:
+            if a._block_type is not None:
+                yield a._block_type
+
+    def _count_num_of_parent_columns(
+        self, anscestor_block_types: AnscestorBlockTypes
+    ) -> int:
+        return sum(
+            1
+            for anscestor_block in anscestor_block_types
+            if anscestor_block == "column"
+        )
```

Comment:
Tagging @raethlein just if there are any opinions here. We talked about some refactors here and want to make sure they align.